primitives/parathread-filter-common/src/lib.rs

```diff
@@ -0,0 +1,36 @@
+// Copyright (C) Moondance Labs Ltd.
+// This file is part of Tanssi.
+
+// Tanssi is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Tanssi is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Tanssi.  If not, see <http://www.gnu.org/licenses/>
+#![cfg_attr(not(feature = "std"), no_std)]
+use {
+    sp_std::marker::PhantomData,
+    tp_traits::{GetSessionContainerChains, ParaId, ParathreadHelper},
+};
+
+// Common implementation of the ParathreadHelper trait for all chains
+// supporting pallet_session and pallet_registrar.
+pub struct ExcludeAllParathreadsFilter<Runtime>(PhantomData<Runtime>);
+impl<Runtime: pallet_session::Config + pallet_registrar::Config> ParathreadHelper
+    for ExcludeAllParathreadsFilter<Runtime>
+{
+    fn is_parathread(para_id: &ParaId) -> bool {
+        pallet_registrar::Pallet::<Runtime>::session_container_chains(
+            pallet_session::Pallet::<Runtime>::current_index().into(),
+        )
+        .parathreads
+        .iter()
+        .any(|(id, _)| id == para_id)
```

Comment:
I believe @girazoki already mentioned it, but its better for this function to take a `&mut Vec<ParaId>` and filter all at once. And actually it may be easier to just return the list of parathreads and do the filtering in the pallet.

Reply:
Refactored in https://github.com/moondance-labs/tanssi/pull/899/commits/14c88e43259860951ca9797cc60e1e845dbd18a2